code/datums/part_customization.dm

```diff
@@ -15,6 +17,8 @@ ABSTRACT_TYPE(/datum/part_customization)
 
 	///Check if we can, then apply the part
 	proc/try_apply(mob/M, list/custom_parts = null)
+		if (src.associated_trait_id && M.traitHolder)
+			M.traitHolder.addTrait(associated_trait_id)
```

Comment:
This should be after the `can_apply` check, either in `apply_to` (with `SHOULD_CALL_PARENT` set) or just above it inside the if statement.

Reply:
opted for the `SHOULD_CALL_PARENT` approach